openpype/hosts/nuke/plugins/publish/validate_asset_context.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+"""Validate if instance asset is the same as context asset."""
+from __future__ import absolute_import
+
+import pyblish.api
+
+import openpype.hosts.nuke.api.lib as nlib
+
+from openpype.pipeline.publish import (
+    RepairAction,
+    ValidateContentsOrder,
+    PublishXmlValidationError,
+    OptionalPyblishPluginMixin,
+    get_errored_instances_from_context
+)
+
+
+class SelectInvalidNodesAction(pyblish.api.Action):
+    """Select invalid nodes."""
+
+    label = "Select Failed Node"
+    icon = "briefcase"
+    on = "failed"
+
+    def process(self, context, plugin):
+        if not hasattr(plugin, "select"):
+            raise RuntimeError("Plug-in does not have repair method.")
+
+        # Get the failed instances
+        self.log.debug("Finding failed plug-ins..")
+        failed_instance = get_errored_instances_from_context(context, plugin)
+        if failed_instance:
+            self.log.debug("Attempting selection ...")
+            plugin.select(failed_instance.pop())
```

Comment:
This is wrong, you are only ever selecting the invalid node for the last instance you should be selecting all.

Reply:
0953dc65cc915bd78cd1b37ba4305ce4fc705aa8